src/eva/data/mon_data_space.py

```diff
@@ -200,87 +196,279 @@ def subset_coordinate(self, ds, coordinate, requested_subset):
+                if line.find('channel=') != -1:
                     strs = line.split()
-                    sensor = strs[1]
-                    satellite = strs[2]
+                    if strs[4].isdigit():
+                        channo.append(int(strs[4]))
 
-            channo = np.array(chans, dtype=int)
-        return channo, nchans, nregion, satellite, sensor
+            # The list of variables is at the end of the file between the lines
+            # "vars" and "end vars".
+            start = len(lines) - (nvars + 1)
+            for x in range(start, start + nvars):
+                strs = lines[x].split()
+                vars.append(strs[-1])
+
+        return coords, dims, attribs, nvars, vars, channo
 
     # ----------------------------------------------------------------------------------------------
 
-    def read_radmon_ieee(self, file_name, nchans, nregion, file_path=None):
-
-        if file_path:
-            filename = os.path.join(file_path, file_name)
-        else:
-            filename = file_name
-        if not os.path.isfile(filename):
-            count = None
-            penalty = None
-            omgnbc_sum = None
-            omgnbc_sum = None
-            total_sum = None
-            omgbc_sum = None
-            omgnbc_sum2 = None
-            omgbc_sum2 = None
-            total_sum2 = None
-            cycle_tm = None
-            return count, penalty, omgnbc_sum, omgnbc_sum, total_sum, \
-                omgbc_sum, omgnbc_sum2, omgbc_sum2, cycle_tm
+    def read_ieee(self, file_name, coords, dims, ndims_used, nvars, vars, file_path=None):
 
         # find cycle time in filename and create cycle_tm as datetime object
         cycle_tm = None
-        cycstrs = filename.split('.')
+        cycstrs = file_name.split('.')
 
         for cycstr in cycstrs:
             if cycstr.isnumeric():
                 cycle_tm = datetime(int(cycstr[0:4]), int(cycstr[4:6]),
                                     int(cycstr[6:8]), int(cycstr[8:]))
 
+        filename = os.path.join(file_path, file_name) if file_path else file_name
+
+        if not os.path.isfile(filename):
+            rtn_array = None
+            return rtn_array
+
         # read binary Fortran file
         f = FortranFile(filename, 'r', '>u4')
 
-        count = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        penalty = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        omgnbc_sum = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        total_sum = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        omgbc_sum = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        omgnbc_sum2 = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        total_sum2 = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
-        omgbc_sum2 = f.read_reals(dtype=np.dtype('>f4')).reshape(nchans, nregion)
+        if ndims_used == 1:
```

Comment:
Can we at least combine these somehow where the if statement defines the empty size and dims that get reshaped? the for loops should (mostly, maybe need a dim=3 version) be able to be written only once

Reply:
Fair point.  I've made it a bit more concise.  The dim=3 case is still a special case, but maybe I can figure a way to make it less special.

Reply:
Sounds like a plan, this is already more concise but if you can make it even better with your new plan, that would be great!